VitalSignsMonitoring/VitalSignsMonitoring/tmatrix.h

```
@@ -27,16 +28,18 @@ namespace vitalSignsMonitoring {
 
 			// This constructor deep copies another matrix
 			TMatrix(const TMatrix<T>& rhs) : Matrix(rhs) {
-				if (!isDataAllocated() || (rhs.m_rows * rhs.m_columns != m_rows * m_columns)) {
-					free();
-					allocate(rhs.m_rows, rhs.m_columns);
+				if (rhs.isDataAllocated()) {
+					if (!isDataAllocated() || (rhs.m_rows * rhs.m_columns != m_rows * m_columns)) {
+						free();
```

If !rhs.isDataAllocated(), maybe should raise an exception.

---

minimal exception handling since we aim for embedded. I'll add a log